### DISCUSSION TITLE
Receive message from SceneSwitcher

### DISCUSSION THREAD

**Discussion by devock** [obs-websocket-community-projects/obs-websocket-js#309] [Q&A]
Hello. I want to receive messages from the OBS plugin SceneSwitcher, but I don't understand witch function i need to use, can you help me ? thanks

**Comment by devock** [comment]
Into logs I have the websocket operation : 

`18:10:16.403: [obs-websocket] [debug] [WebSocketServer::onMessage] Incoming message (decoded):
18:10:16.403: {
18:10:16.403:   "d": {
18:10:16.403:     "requestData": {
18:10:16.403:       "requestData": {
18:10:16.403:         "message": "testchange"
18:10:16.403:       },
18:10:16.403:       "requestType": "AdvancedSceneSwitcherMessage",
18:10:16.403:       "vendorName": "AdvancedSceneSwitcher"
18:10:16.403:     },
18:10:16.403:     "requestId": "testchange - ws://localhost:4455",
18:10:16.403:     "requestType": "CallVendorRequest"
18:10:16.403:   },
18:10:16.403:   "op": 6
18:10:16.403: }
18:10:16.403: [obs-websocket] [debug] [WebSocketServer::onMessage] Outgoing message:
18:10:16.403: {
18:10:16.403:   "d": {
18:10:16.403:     "requestId": "testchange - ws://localhost:4455",
18:10:16.403:     "requestStatus": {
18:10:16.403:       "code": 100,
18:10:16.403:       "result": true
18:10:16.403:     },
18:10:16.403:     "requestType": "CallVendorRequest",
18:10:16.403:     "responseData": {
18:10:16.403:       "requestType": "AdvancedSceneSwitcherMessage",
18:10:16.403:       "responseData": {},
18:10:16.403:       "vendorName": "AdvancedSceneSwitcher"
18:10:16.403:     }
18:10:16.403:   },
18:10:16.403:   "op": 7
18:10:16.403: }`

But I don't intercept it width VendorEvent : 

`obs.on('VendorEvent', ({vendorName, eventType, eventData}) => {
            console.log(eventType)
        });`

**Reply by t2t2** [reply]
That log doesn't include any events being sent, only obs-websocket receiving a `CallVendorRequest` request that it responds to
